tests/integration/cmd_devfile_push_test.go

```diff
@@ -0,0 +1,156 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odo devfile push command tests", func() {
+	var project string
+	var context string
+	var currentWorkingDirectory string
+
+	var sourcePath = "/projects"
+
+	oc = helper.NewOcRunner("oc")
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		project = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		currentWorkingDirectory = helper.Getwd()
+		helper.Chdir(context)
+		os.Setenv("ODO_EXPERIMENTAL", "true")
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		helper.Chdir(currentWorkingDirectory)
+		os.Unsetenv("ODO_EXPERIMENTAL")
+	})
+
+	Context("Verify devfile push works", func() {
+
+		It("Check that odo push works with a devfile", func() {
+			project = helper.CreateRandProject()
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+
+			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", project)
+			Expect(output).To(ContainSubstring("✓  Waiting for component to start"))
+			Expect(output).To(ContainSubstring("✓  Push devfile component"))
+			Expect(output).To(ContainSubstring("✓  Changes successfully pushed to component"))
```

Comment:
One assertion that confirms the devfile push is successful would be enough. So  ```Expect(output).To(ContainSubstring("Changes successfully pushed to component"))``` would be the proper fit imo. Do not to pass character like ```✓``` in assert as it may behave differently on terminal like PowerShell.

Reply:
Ah, right good point.